### DISCUSSION TITLE
Can you use PNP Search with Query Strings and Page Query Variables to create dynamic rollup pages?

### DISCUSSION THREAD

**Discussion by ElephantLathe** [microsoft-search/pnp-modern-search#2582] [Q&A] [answered]
We're migrating our on prem 2013 SharePoint Intranet into SharePoint Online (SPO) using the modern site experience and I'm trying to backfill features that seemed to have disappeared in SPO. I have spent quite a bit of time trying to find out if PNP Search can do the two things below, but I've not been able to find a definitive answer or walkthrough yet, I'd really appreciate a pointer to something that can help me achieve this, or if it's not currently possible for someone to put me out of misery trying to find out whether it is.

We use MMD to tag all content on our Intranet. I'm looking for three things that provide dynamic rollup of content that we used to be able to do in classic SharePoint sites.

1. We used query strings along with a single page that showed a rollup of results when someone clicked a 'View All' link. As long as the content editor added a query string with the MMD term when they linked to that page it worked and meant they didn't have to configure any web parts every time they wanted to show a rollup. Is this possible in PNP Search, and how might we configure that?
2. We used to have publishing page template that showed related content using the Content Query web part. This looked at the page query variable (e.g. in classic it was [PageFieldValue: Intranet Tags]), so whatever the tag used on the page that was created it pulled in any articles tagged the same, is that possible with PNP Search and how might we configure that? This link references query variables but also notes not everything may work (https://learn.microsoft.com/en-us/sharepoint/technical-reference/query-variables)

Hope that makes sense.

Thanks,
Jonathan

**Comment by ElephantLathe** [comment, marked as answer]
I think I've found the answer for 1, a lot of it was not understanding it all relates to what is called 'Tokens' in the documentation https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/ there are some examples of exactly what I needed to do in there. Using this in the query template section in PNP Search config works e.g. {|owstaxidmetadataalltagsinfo:{Page.IntranetCategories.Label}} then you can save your site page as a site page template and then when reused, it pulls in results based on the tag the new page is given.


Solved 2 as well regarding query strings

Query String Example 

add ?Tag=MyTagTerm to the URL of the destination page with the PNP Search Web part configured.

This code in the 'Query Template' section of the Search Results web part searches for everything with that Query String Tag

**Reply by wobba** [reply]
Do you mean pick up page metadata into the link which then goes to another page?
